src/metrics/matcher/config.go

```diff
@@ -114,6 +117,12 @@ func (cfg *Configuration) NewOptions(
 		NameAndTagsFn:       m3.NameAndTags,
 		SortedTagIteratorFn: sortedTagIteratorFn,
 	}
+	includeTagKeys := make(map[uint64]struct{}, len(cfg.IncludeTagKeys))
```

Comment:
Nit: please extract this block of code into a separate function that takes `[]string` and returns `map[uint64]struct{}`